### PR TITLE
admin global message

### PR DESCRIPTION
**Pull request by tuanh118**
Admin
- Add change room dropdown
- Add send all checkbox

### CODE REVIEW

File: frontend/src/Components/Chatbox.tsx [hunk 1]
```diff
@@ -61,27 +54,20 @@ function ChatMessage({ message_type, payload, action }: Message) {
...
   const [messageInput, setMessageInput] = React.useState("");
   const [rooms, setRooms] = React.useState([] as Room[]);
+  const [sendAll, setSendAll] = React.useState(false);
   const [show, setShow] = React.useState(false);
   const socket = useSocket();
   const userData = useUserData();
 
-  const loadRooms = (event: any) => {
-    event.preventDefault();
+  const isAdmin = userData?.isLoggedIn && userData?.role == 0;
```
Comment (namdaoduy) [inline frontend/src/Components/Chatbox.tsx:68]: We should use `===` instead of `==`

File: frontend/src/Components/FunctionButtonGroup.tsx [hunk 2]
```diff
@@ -1,98 +1,109 @@
...
-
-  const [switchRoomOpen, setSwitchRoomOpen] = React.useState(false);
-  const handleSwitchRoomOpen = () => {
-    setSwitchRoomOpen(true);
-  };
-  const handleSwitchRoomClose = () => {
-    setSwitchRoomOpen(false);
-  };
+  const onSwitchRoomConfirm = () => {
+    // Start timer after successful room change
```
Comment (namdaoduy) [inline frontend/src/Components/FunctionButtonGroup.tsx:24]: Please add `TODO` tag

File: frontend/src/Types/User.tsx [hunk 3]
```diff
@@ -1,9 +1,7 @@
-
-
 export type UserData = {
     isLoggedIn: boolean,
     jwtToken?: string,
     username?: string,
-    role?: number
+    role?: number // admin = 0, mod = 1, member = 2
```
Comment (namdaoduy) [inline frontend/src/Types/User.tsx:5]: TODO: use constants for Roles

**Review by namdaoduy — CHANGES_REQUESTED** [review]

File: frontend/src/Types/User.tsx (hunk 3 above)
Comment (vudung45) [reply frontend/src/Types/User.tsx:5]: Ideally this should be enum (same on server), but yeah let's add that as TODO

Comment (namdaoduy) [reply frontend/src/Types/User.tsx:5]: @vudung45 I think Enum is not as reliable as string/int constants. Change order of the roles and everything mess up. We need to use it carefully, or we can assign value for Enum, right?

**Review by namdaoduy — APPROVED** [review]

File: frontend/src/Types/User.tsx (hunk 3 above)
Comment (vudung45) [reply frontend/src/Types/User.tsx:5]: Yeah, i think it's better to leave to have them consts, you brought up a good point about changing the order of enum :D. Let's do that in the next PR 🗡️ 
The best practice is probably to introduce a way to synchronize this between server and client.

Comment (manhhung741) [reply frontend/src/Types/User.tsx:5]: Bọn em có thể dùng string làm enum nhé' https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

**Pull request merged**